src/components/modal/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { FC, PropsWithChildren } from 'react';
+import {
+  Modal as ChakraModal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  ModalProps,
+} from '@chakra-ui/react';
+
+import { H3 } from '../heading';
+import Divider from '../divider';
+import Button from '../button';
+
+type ActionButton = {
+  action: () => void;
+  label: string;
+};
+
+interface Props
+  extends Omit<ModalProps, 'isCentered' | 'motionPreset' | 'size'> {
```

Comment:
as per https://automotivesmg.slack.com/archives/C03GK939L3V/p1671548157757039, the general agreement would be to be restrictive. Therefore I'd change this to be: 
```suggestion
  extends Pick<ModalProps, 'onClose' | 'isOpen'> {
```